R/updateRecodeDB.R

```diff
@@ -0,0 +1,48 @@
+#' Update Recode Data Base
+#'
+#' Update a recode data base.
+#'
+#'@param newRecodes A `data.frame` containing new recode information.
+#'@param recodeDBPath Path to the `.xlsx` file in which the data base is stored.
+#'@param newRecodeDBPath Path to the `.xlsx` file in which the updated data base should be stored.
+#'@param name Name of the specific recode list.
+#'@param override Logical of length 1. Should existing recode pairs be overwritten when conflicting newer recode pairs
+#'are present in the `newRecodes`?
+#'
+#'@return NULL
+#'
+#'
+#'@examples
+#'# tbd
+#'
+#'@export
+updateRecodeDB <- function(newRecodes, recodeDBPath, newRecodeDBPath, name, override = FALSE) {
+  recode_db <- getRecodeDB(filePath = recodeDBPath)
+  old_recode_list <- recode_db[[name]]
+  checkRecodeList(old_recode_list)
+
+  oldValues_conflicts <- character()
+  if(override) {
+    newRecodes_manual <- unique(newRecodes)
+    oldValues_conflicts <- newRecodes_manual[newRecodes_manual$oldValues %in% old_recode_list$oldValues, "oldValues"]
+    if(length(oldValues_conflicts) > 0) {
+      message("The recodes for the following oldValues in the existing data base in sheet '", name, "' will be overwritten: ",
+              paste(oldValues_conflicts, collapse = ", "))
```

Comment:
Maybe we could also add the values `oldValues_conflicts` will be overwritten with?

Reply:
I agree that this is valuable information. I have extended the messages to the old `newValues` and the new `newValues` and restructured the function a bit to achieve this. Let me know what you think!

Reply:
Looks great now, thanks for implementing!